### PR TITLE
InfluxDB: exclude result and table column from Flux table results

### PR DESCRIPTION
**Pull request by ryantxu**
Extracted from #26634

For non-timeseries results, we current include two useless columns: "results" and "table."  This PR removes them:

image

### CODE REVIEW

File: pkg/tsdb/influxdb/flux/executor.go [hunk 1]
```diff
@@ -20,29 +20,25 @@ func executeQuery(ctx context.Context, query queryModel, runner queryRunner, max
...
+	// Make sure there is at least one frame
+	if len(dr.Frames) < 1 {
+		dr.Frames = append(dr.Frames, data.NewFrame(""))
 	}
-
+	firstFrame := dr.Frames[0]
+	if firstFrame.Meta == nil {
+		firstFrame.SetMeta(&data.FrameMeta{})
+	}
+	firstFrame.Meta.ExecutedQueryString = flux
```
Comment (ryantxu) [inline pkg/tsdb/influxdb/flux/executor.go:41]: These changes make sure we always return at least one frame, and the frame has the `ExecutedQueryString` set.

Previously, this was only attached when `runQuery` had an error, not when readDataFrames produces an error (or no results)

File: pkg/tsdb/influxdb/flux/executor_test.go [hunk 2]
```diff
@@ -270,3 +269,34 @@ func TestGoldenFiles(t *testing.T) {
 		}
 	})
 }
+
+func TestRealQuery(t *testing.T) {
+	t.Skip() // this is used for local testing
```
Comment (ryantxu) [inline pkg/tsdb/influxdb/flux/executor_test.go:274]: @aknuds1 -- thoughts on how/if we should include this?

I added this so I could easily start the debugger and step though execution process locally.

Comment (aknuds1) [reply pkg/tsdb/influxdb/flux/executor_test.go:274]: I will have to check,when time allows.

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Just a quick review for now, but left a suggestion for an error message.

File: pkg/tsdb/influxdb/flux/executor_test.go [hunk 3]
```diff
@@ -270,3 +269,34 @@ func TestGoldenFiles(t *testing.T) {
...
+			JsonData: json,
+			SecureJsonData: securejsondata.GetEncryptedJsonData(map[string]string{
+				"token": "PjSEcM5oWhqg2eI6IXcqYJFe5UbMM_xt-UNlAL0BRYJqLeVpcdMWidiPfWxGhu4Xrh6wioRR-CiadCg-ady68Q==",
+			}),
+		}
+
+		runner, err := runnerFromDataSource(dsInfo)
+		if err != nil {
+			t.Fatal(err)
+		}
```
Comment (aknuds1) [inline pkg/tsdb/influxdb/flux/executor_test.go:291]:
```suggestion
		require.NoError(t, err)
```

File: pkg/tsdb/influxdb/flux/executor_test.go [hunk 4]
```diff
@@ -270,3 +269,34 @@ func TestGoldenFiles(t *testing.T) {
...
+
+		dr := executeQuery(context.Background(), queryModel{
+			MaxDataPoints: 100,
+			RawQuery:      "buckets()",
+		}, runner, 50)
+		err = experimental.CheckGoldenDataResponse("./testdata/buckets-real.golden.txt", &dr, true)
+		if err != nil {
+			t.Fatal(err)
+		}
```
Comment (aknuds1) [inline pkg/tsdb/influxdb/flux/executor_test.go:300]:
```suggestion
		require.NoError(t, err)
```

File: pkg/tsdb/influxdb/flux/executor_test.go (hunk 2 above)
Comment (aknuds1) [reply pkg/tsdb/influxdb/flux/executor_test.go:274]: I guess it's OK to keep it like this :shrug:

File: pkg/tsdb/influxdb/flux/executor.go [hunk 5]
```diff
@@ -20,29 +20,25 @@ func executeQuery(ctx context.Context, query queryModel, runner queryRunner, max
...
 
-	dr = readDataFrames(tables, int(float64(query.MaxDataPoints)*1.5), maxSeries)
-
-	for _, frame := range dr.Frames {
-		if frame.Meta == nil {
-			frame.Meta = &data.FrameMeta{}
-		}
-		frame.Meta.ExecutedQueryString = flux
+	// Make sure there is at least one frame
+	if len(dr.Frames) < 1 {
```
Comment (aknuds1) [inline pkg/tsdb/influxdb/flux/executor.go:34]: I think this is more idiomatic and reads clearer:
```suggestion
	if len(dr.Frames) == 0 {
```

**Review by aknuds1 — APPROVED** [review]
LGTM, but see suggestions.

**Comment by ryantxu** [comment]
Fixes #26927

**Pull request merged**